### PR TITLE
[ecosystem] Use nginx + passenger to run the app in production.

### PR DESCRIPTION
**Pull request by zacharydenton**
## Motivation

Updates the prod configuration to use nginx + passenger instead of puma with nothing in front of it. This frees up Rails from serving static assets and enables compression. It should also speed up deploys as the passenger-full base image already has the dependencies we need.

## Test Plan

Tested manually

## Related PRs

(If this PR adds or changes functionality, please take some time to update the docs at https://github.com/aptos-labs/aptos-core/tree/main/developer-docs-site, and link to your PR here.)

### CODE REVIEW

**Comment by geekflyer** [comment]
hum, we run the community app in AWS App Runner in an autoscaling container context and there is already a load balancer in front of it. We're also going to switch to GCP Cloud Run soon which is effectively managed k8s with a load balancer in front of it. Is passenger and nginx really useful/required in that context? 
A cursory quick research seems like in a multi-server/multi-container context with an external load balancer in front of the servers/containers, most of the benefits of passenger and in particular nginx seem to go away. The passenger docs even sort of indicate so themselves: https://www.phusionpassenger.com/docs/advanced_guides/config_and_optimization/standalone/optimization.html#switching-to-passenger-standalone-with-the-builtin-http-engine states:
> In some multi-server setups, Nginx and Apache may be redundant. Recall that in typical multi-server setups there is a load balancer which forwards requests to one of the many web servers. Each web server in this setup runs Passenger. But the load balancer is sometimes already responsible for many of the tasks that Nginx and Apache perform, e.g. the secure handling of HTTP connections, buffering, slow client protection or even static file serving.

I understand we want to have gzip compression, http>https redirects and offload static file serving.
We can actually just configure that on the load balancer level in GCP Cloud Run. I can just add that.
On static file serving I can just configure our infra such that we run a separate container that only handles the static file serving (we can probably use the same image).
Fyi the http->https redirect is a AWS AppRunner specific limitation that we won't be able to workaround with passenger since AppRunners load balancer doesn't even accept http traffic so it won't reach nginx/apache ever.
With GCP Cloud Run this is solved though.

Anyways, long story short. I'm mostly concerned by the complexity/duplication of introducing nginx and seperate nginx config files here + the additional overhead of each request going through a container-internal nginx. I think it might be fine to use passenger in standalone mode or with the built-in http server, since this probably won't add much complexity but makes the image build looking slightly simpler.

**Comment by zacharydenton** [comment]
@geekflyer The main thing is offloading Rails from serving static files (since it can't serve dynamic requests when it's serving a static file). I don't think the load balancer can handle this, at least not the one we're currently using (https://github.com/envoyproxy/envoy/issues/378). Instead of setting up a separate container that serves the static files, I would prefer that we instead set up a CDN that serves the static files (and potentially also handles compression, http->https, etc.). What you see here is just a stopgap solution until we have a CDN.

File: ecosystem/platform/server/Dockerfile [hunk 1]
```diff
@@ -7,48 +7,41 @@ RUN git clone --depth=1 --branch v1.3.0 https://github.com/aptos-labs/aws-env sr
...
 ENV SECRET_KEY_BASE=18ff7d41b9da02d394434e2eb140611516707334b7d7a7e15cf8b567061f30e9400a82e61aa772b9e6ccb72853932769d03bbbb9d78f62333c8f7adb95cc727d
 
 # Yes, this should run twice
 # https://github.com/rails/tailwindcss-rails/issues/158
 RUN SKIP_DB_CHECK=1 bundle exec rake assets:precompile && SKIP_DB_CHECK=1 bundle exec rake assets:precompile
 
-COPY --link --from=aws-env /aws-env /aws-env
-
+USER root
+CMD ["/aws-env", "exec", "--", "/sbin/my_init"]
```
Comment (CapCap) [inline ecosystem/platform/server/Dockerfile:46]: lets remove anything cloud specific from our containers; worst case we override the CMD when running the image it from a cloud context

Comment (zacharydenton) [reply ecosystem/platform/server/Dockerfile:46]: Should we also remove it from the current image?

https://github.com/aptos-labs/aptos-core/blob/93f81c4a34978daf29e1fef54185ea86be0392a7/ecosystem/platform/server/Dockerfile#L54

Comment (geekflyer) [reply ecosystem/platform/server/Dockerfile:46]: don't remove it just yet. This is required for the container to be able to load secrets on AWS AppRunner which doesn't support secret injection natively. Once we move to GCP we can remove this.

**Comment by geekflyer** [comment]
> @geekflyer The main thing is offloading Rails from serving static files (since it can't serve dynamic requests when it's serving a static file). I don't think the load balancer can handle this, at least not the one we're currently using ([envoyproxy/envoy#378](https://github.com/envoyproxy/envoy/issues/378)). Instead of setting up a separate container that serves the static files, I would prefer that we instead set up a CDN that serves the static files (and potentially also handles compression, http->https, etc.). What you see here is just a stopgap solution until we have a CDN.

Ok, let's just set up a CDN then. I can do that after the GCP migration this week.

File: ecosystem/platform/server/Dockerfile (hunk 1 above)
Comment (zacharydenton) [reply ecosystem/platform/server/Dockerfile]: Done.

File: ecosystem/platform/server/Dockerfile [hunk 2]
```diff
@@ -1,54 +1,38 @@
...
 
-# build aws-env which is used to inject load AWS Secrets as environment variables at init time.
-RUN git clone --depth=1 --branch v1.3.0 https://github.com/aptos-labs/aws-env src \
-    && cd src/cmd/aws-env \
-    && CGO_ENABLED=0 GOARCH=amd64 GOOS=linux go build -ldflags '-w -s' -o /aws-env && chmod +x /aws-env
+RUN npm install -g yarn
 
-FROM ruby:3.1.2-slim
+RUN rm -f /etc/service/nginx/down
+RUN rm -f /etc/nginx/sites-enabled/default
```
Comment (CapCap) [inline ecosystem/platform/server/Dockerfile:8]:
```
RUN npm install -g yarn && \
         rm -f /etc/service/nginx/down
         rm -f /etc/nginx/sites-enabled/default
```

Comment (zacharydenton) [reply ecosystem/platform/server/Dockerfile]: Done.

File: ecosystem/platform/server/nginx.conf [hunk 3]
```diff
@@ -0,0 +1,30 @@
+server {
+    listen 3000;
+    server_name community.aptoslabs.com;
```
Comment (geekflyer) [inline ecosystem/platform/server/nginx.conf:3]: how would this work in other environments, like staging and dev? Can we override this somehow? E.g. consider https://community.staging.gcp.aptosdev.com/

**Review by geekflyer — APPROVED** [review]
pre-approving this now, but please address my comment/question on how we can override server_name in staging/dev environments.

File: ecosystem/platform/server/nginx.conf (hunk 3 above)
Comment (zacharydenton) [reply ecosystem/platform/server/nginx.conf]: Updated to handle all connections to port 3000.

**Pull request closed without merging**